chia/wallet/puzzles/vault_p2_recovery.clsp

```diff
@@ -62,5 +62,4 @@
     (list ASSERT_MY_AMOUNT my_amount)
```

Comment:
The singleton top layer already puts an ASSERT_MY_AMOUNT condition so we end up with a redundant assertion.

Reply:
Yes, but unfortunately, we don't know from the inner puzzle here that the singleton has already done that.  We are being passed an amount in the solution and while someone may tell the truth at the singleton layer, they may lie to us at this layer, so we have to assert the amount all the same.